pkgs/by-name/gi/gitify/package.nix

```diff
@@ -0,0 +1,109 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  pnpm,
+  nodejs,
+  electron,
+  makeDesktopItem,
+  copyDesktopItems,
+  imagemagick,
+  makeWrapper,
+  nix-update-script,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "gitify";
+  version = "5.16.1";
+
+  src = fetchFromGitHub {
+    owner = "gitify-app";
+    repo = "gitify";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-mTe3Nl0ZxUfzS06oETOh/gdcjVWeXSoyXmvKKHJ2rVw=";
+  };
+
+  nativeBuildInputs = [
+    nodejs
+    pnpm.configHook
+    copyDesktopItems
+    imagemagick
+    makeWrapper
+  ];
+
+  pnpmDeps = pnpm.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    hash = "sha256-xgS7MaiILErKHCSI/f2lQif8Hf3RPQEC/DGGGmOBRzk=";
+  };
+
+  env.ELECTRON_SKIP_BINARY_DOWNLOAD = 1;
+
+  buildPhase = ''
+    runHook preBuild
+
+    # electronDist needs to be modifiable on Darwin
+    cp -r ${electron.dist} electron-dist
+    chmod -R u+w electron-dist
+
+    pnpm build
+    pnpm exec electron-builder \
+        --dir \
+        -c.electronDist=electron-dist \
+        -c.electronVersion="${electron.version}" \
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    ${
+      if stdenv.isDarwin then
+        ''
+          mkdir -p $out/Applications
+          cp -r dist/mac*/Gitify.app $out/Applications
+          makeWrapper $out/Applications/Gitify.app/Contents/MacOS/gitify $out/bin/gitify
+        ''
+      else
+        ''
+          mkdir -p $out/share/gitify
+          cp -r dist/*-unpacked/{locales,resources{,.pak}} $out/share/gitify
+
+          for size in 16 32 48 64 128 256 512 1024; do
+            mkdir -p $out/share/icons/hicolor/"$size"x"$size"/apps
+            magick assets/images/app-icon.ico -resize "$size"x"$size" $out/share/icons/hicolor/"$size"x"$size"/apps/gitify.png
+          done
```

Comment:
I think it is enough to install one size

Reply:
Thanks for the review!
I will leave only the size of the original icon, which is 256x256.